### PR TITLE
Test suite fix for batchPD HIP

### PR DESCRIPTION
**Pull request by r-abishek**
Adds the hipDeviceSynchronize() after all hip function calls.

### CODE REVIEW

File: utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp [hunk 1]
```diff
@@ -765,6 +765,8 @@ int main(int argc, char **argv)
         else
             missingFuncFlag = 1;
 
+        hipDeviceSynchronize();
```
Comment (rrawther) [inline utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp:768]: Please move it outside the switch statement so you need to only make one line code change. Applies to all the files in this commit

**Review by rrawther — COMMENTED** [review]
I think, you can move the hipDeviceSync.. () call outside the switch statement to have minimal code change.

File: utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp (hunk 1 above)
Comment (r-abishek) [reply utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp:768]: @rrawther The hipDeviceSynchronize() actually needs to be within the clock() timed section (before end = clock(); is called). Only then does the clock() wait for the GPU-offloads to finish entirely, and the end time can be recorded.

Comment (r-abishek) [reply utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp:768]: https://github.com/r-abishek/rpp/blob/ar/test_suite_mods/utilities/rpp-unittests/HIP_NEW/Tensor_hip_pkd3.cpp#L577-L598

Comment (rrawther) [reply utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp:768]: probably you can move both outside the switch statement in that case. It is better to keep common code all at one place

Comment (r-abishek) [reply utilities/rpp-unittests/HIP_NEW/BatchPD_hip_pln1.cpp]: Ive now changed all instances in all test suite files to reflect this change. The end = clock() for HOST and the hipDeviceSynchronize() + end = clock() for HIP have been moved outside the switch case

**Review by rrawther — APPROVED** [review]

**Pull request merged**